plugins/woocommerce/client/blocks/assets/js/base/context/hooks/cart/use-store-cart.ts

```diff
@@ -122,122 +121,113 @@ export const defaultCartData: StoreCart = {
-			const cartIsLoading =
-				// @ts-expect-error `hasFinishedResolution` is not typed in @wordpress/data yet.
-				! store.hasFinishedResolution( 'getCartData' );
-
-			const isLoadingRates = store.isCustomerDataUpdating();
-			const { receiveCart, receiveCartContents } = dispatch( cartStore );
-
-			const cartFees =
-				cartData.fees.length > 0
-					? cartData.fees.map( ( fee: CartResponseFeeItem ) =>
-							decodeValues( fee )
-					  )
-					: EMPTY_CART_FEES;
-
-			// Add a text property to the coupon to allow extensions to modify
-			// the text used to display the coupon, without affecting the
-			// functionality when it comes to removing the coupon.
-			const cartCoupons: CartResponseCoupons =
-				cartData.coupons.length > 0
-					? cartData.coupons.map(
-							( coupon: CartResponseCouponItem ) => ( {
-								...coupon,
-								label: coupon.code,
-							} )
-					  )
-					: EMPTY_CART_COUPONS;
-
-			// Update refs to keep the hook stable.
-			const billingAddress = emptyHiddenAddressFields(
-				decodeValues( cartData.billingAddress )
-			);
-			const shippingAddress = cartData.needsShipping
-				? emptyHiddenAddressFields(
-						decodeValues( cartData.shippingAddress )
-				  )
-				: billingAddress;
-
-			if (
-				! fastDeepEqual( billingAddress, billingAddressRef.current )
-			) {
-				billingAddressRef.current = billingAddress;
-			}
-
-			if (
-				! fastDeepEqual( shippingAddress, shippingAddressRef.current )
-			) {
-				shippingAddressRef.current = shippingAddress;
-			}
-
-			return {
-				cartCoupons,
-				cartItems: cartData.items,
-				crossSellsProducts: cartData.crossSells,
-				cartFees,
-				cartItemsCount: cartData.itemsCount,
-				cartItemsWeight: cartData.itemsWeight,
-				cartNeedsPayment: cartData.needsPayment,
-				cartNeedsShipping: cartData.needsShipping,
-				cartItemErrors: cartData.errors,
-				cartTotals,
-				cartIsLoading,
-				cartErrors,
-				billingData: billingAddressRef.current,
-				billingAddress: billingAddressRef.current,
-				shippingAddress: shippingAddressRef.current,
-				extensions: cartData.extensions,
-				shippingRates: cartData.shippingRates,
-				isLoadingRates,
-				cartHasCalculatedShipping: cartData.hasCalculatedShipping,
-				paymentRequirements: cartData.paymentRequirements,
-				receiveCart,
-				receiveCartContents,
-			};
-		},
-		[ shouldSelect ]
+	const { receiveCart, receiveCartContents } = useDispatch( cartStore );
+	const { cartData, cartErrors, cartTotals, cartIsLoading, isLoadingRates } =
+		useSelect(
+			( select ) => {
+				const store = select( cartStore );
+				return {
+					cartData: store.getCartData(),
+					cartErrors: store.getCartErrors(),
+					cartTotals: store.getCartTotals(),
+					cartIsLoading:
+						// @ts-expect-error `hasFinishedResolution` is not typed in @wordpress/data yet.
+						! store.hasFinishedResolution( 'getCartData' ),
+					isLoadingRates: store.isCustomerDataUpdating(),
+				};
+			},
+			[ shouldSelect ]
```

Comment:
Is this actually doing anything useful here? As I understand it, this would only recreate the selector function if `shouldSelect` changes, I think it's fine to keep the deps array empty here since `shouldSelect` doesn't affect the outcome of the function.

Reply:
https://github.com/woocommerce/woocommerce/pull/57673/commits/ddae31f21f3a13a65fabbeabec7d06092b1144bc